### PR TITLE
Hw3

### PR DESCRIPTION
**Pull request by frostpv**

### CODE REVIEW

File: src/main/java/com/internet/shop/dao/GenericDao.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.internet.shop.dao;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface GenericDao<T, V> {
+    T create(T element);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/GenericDao.java]:
```suggestion
    T create(T item);
```
keep your code consistent, if you use "item" in service, here should be same naming

File: src/main/java/com/internet/shop/services/ShoppingCartServiceImpl.java [hunk 2]
```diff
@@ -17,6 +19,26 @@ public ShoppingCart create(ShoppingCart shoppingCart) {
...
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart item) {
+        return null;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return false;
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/services/ShoppingCartServiceImpl.java]: why these methods are without implementation?

File: src/main/java/com/internet/shop/services/ShoppingCartServiceImpl.java [hunk 3]
```diff
@@ -17,6 +19,26 @@ public ShoppingCart create(ShoppingCart shoppingCart) {
...
+        return null;
+    }
+
+    @Override
+    public List<ShoppingCart> getAll() {
+        return null;
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart item) {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/services/ShoppingCartServiceImpl.java]:
```suggestion
    public ShoppingCart update(ShoppingCart shoppingCart) {
```

File: src/main/java/com/internet/shop/services/OrderServiceImpl.java [hunk 4]
```diff
@@ -40,6 +45,11 @@ public Order get(Long id) {
         return orderDao.getAll();
     }
 
+    @Override
+    public Order update(Order item) {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/services/OrderServiceImpl.java]:
```suggestion
    public Order update(Order order) {
```

File: src/main/java/com/internet/shop/services/ShoppingCartServiceImpl.java (hunk 3 above)
Comment (frostpv) [reply src/main/java/com/internet/shop/services/ShoppingCartServiceImpl.java]: fixed

File: src/main/java/com/internet/shop/services/ShoppingCartServiceImpl.java (hunk 2 above)
Comment (frostpv) [reply src/main/java/com/internet/shop/services/ShoppingCartServiceImpl.java]: fixed

File: src/main/java/com/internet/shop/dao/GenericDao.java (hunk 1 above)
Comment (frostpv) [reply src/main/java/com/internet/shop/dao/GenericDao.java]: fixed

File: src/main/java/com/internet/shop/services/OrderServiceImpl.java (hunk 4 above)
Comment (frostpv) [reply src/main/java/com/internet/shop/services/OrderServiceImpl.java]: fixed

File: src/main/java/com/internet/shop/services/ShoppingCartServiceImpl.java [hunk 5]
```diff
@@ -17,6 +18,36 @@ public ShoppingCart create(ShoppingCart shoppingCart) {
...
+    }
+
+    @Override
+    public boolean delete(ShoppingCart shoppingCart) {
+        return shoppingCartDao.delete(shoppingCart.getId());
+    }
+
+    @Override
+    public ShoppingCart getByUserId(Long userId) {
+        return shoppingCartDao.get(userId).orElseThrow();
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/services/ShoppingCartServiceImpl.java]:
```suggestion
        return shoppingCartDao.get(userId).get();
```

**Review by RomanDubovskyi — APPROVED** [review]
don't forget to change orElseThrow() to get(), left comment

File: src/main/java/com/internet/shop/services/ShoppingCartServiceImpl.java (hunk 5 above)
Comment (frostpv) [reply src/main/java/com/internet/shop/services/ShoppingCartServiceImpl.java]: thanks, fixed

**Pull request merged**